cfme/tests/cloud/test_cloud_proxy.py

```diff
@@ -0,0 +1,149 @@
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, [provider.type], provider)
+
+    settings_local(appliance, [provider.type], True)
+    validate_provider(appliance, provider)
+    flash.assert_message_match('Credential validation was successful')
+
+
+@pytest.mark.uncollectif(lambda provider: provider.type == 'openstack')
+def test_proxy_invalid(appliance, provider, setup_provider, request):
+    """Configures the specific invalid proxy for each provider"""
+    logger.info("Begin Test Invalid Proxy for Provider {}".format(provider.type))
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, [provider.type], provider)
+
+    settings_local(appliance, [provider.type], False)
+    restart_appliance(appliance)
+    with error.expected('Credential validation was not successful:'):
+        validate_provider(appliance, provider)
+        flash.assert_message_match('Credential validation was successful')
+
+
+@pytest.mark.uncollectif(lambda provider: provider.type == 'openstack')
+def test_proxy_remove(appliance, provider, setup_provider, request):
+    """Removes the specific valid proxy for all providers"""
+    logger.info("Begin Test to Remove Proxy")
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, [provider.type], provider)
+
+    appliance.ssh_client.run_command(
+        "rm -f /var/www/miq/vmdb/config/settings.local.yml")
+    restart_appliance(appliance)
+    validate_provider(appliance, provider)
+    flash.assert_message_match('Credential validation was successful')
+
+
+@pytest.mark.uncollectif(lambda provider: provider.type == 'openstack')
+def test_proxy_override(appliance, provider, setup_provider, request):
+    """Configures a bad default and then a valid proxy for each provider"""
+    logger.info("Begin Test Override Proxy for Provider {}".format(provider.type))
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, ['default', provider.type], provider)
+
+    logger.info("Test to fail first")
+    settings_local(appliance, ["default"], False)
+    restart_appliance(appliance)
+    with error.expected('Credential validation was not successful:'):
+        validate_provider(appliance, provider)
+        flash.assert_message_match('Credential validation was successful')
+
+    logger.info("Then test to pass using provider specific override")
+    settings_local(appliance, ['default', provider.type], True)
+    restart_appliance(appliance)
+    validate_provider(appliance, provider)
+    flash.assert_message_match('Credential validation was successful')
+
+
+def settings_local(appliance, proxy_type, valid=True):
+    """Creates the file string and sends it to the appliance"""
+    proxy = cfme_data['proxy_servers']['default']
+    creds = credentials[proxy['credentials']]
+    port = proxy['port'] if valid else '8686'
```

Comment:
I think you missed the point of the suggestion - I understood it was overloading port, I'm just suggesting that there's no need for two variables here. You can just use `port`, once, in the for loop at line 119, and stick this statement in the 'else' clause.